RepresentsMe/Sources/Utilities/LocationManager.swift

```diff
@@ -30,6 +44,57 @@ class LocationManager: NSObject, CLLocationManagerDelegate {
         locationManager.desiredAccuracy = kCLLocationAccuracyBest
     }
 
+    // MARK: - Methods
+
+    /// Adds a listener for changes in location authorization
+    ///
+    /// - Parameter listener:   the LocationAuthorizationListener to add
+    static func addLocationAuthorizationListener(_ listener:LocationAuthorizationListener) {
+        locationAuthorizationListeners.append(listener)
+    }
+
+    /// Removes a listener for changes in location authorization
+    ///
+    /// - Parameter listener:   the LocationAuthorizationListener to remove
+    static func removeLocationAuthorizationListener(_ listener: LocationAuthorizationListener) {
+        if let index = find(listener: listener, in: locationAuthorizationListeners) {
+            locationAuthorizationListeners.remove(at: index)
+        }
+    }
+
+    /// Finds the given listener in the given Array of listeners
+    ///
+    /// - Parameter listener:   the listener to search for
+    /// - Parameter in:         the Array to search
+    ///
+    /// - Returns: the index if found or nil if not found
+    private static func find(listener:LocationAuthorizationListener,
+                             in listeners:[LocationAuthorizationListener]) -> Int? {
+        return listeners.firstIndex {(appStateListener) in
+            if let listener1 = listener as? UIViewController,
+                let listener2 = appStateListener as? UIViewController {
+                return listener1 === listener2
+            }
+
+            // should never occur
+            return false
+        }
+    }
+
+    /// Returns whether location services are available to this app
+    func locationServicesEnabledAndAuthorized() -> Bool {
+        // Check if Location Services are enabled globally
+        if CLLocationManager.locationServicesEnabled() {
```

Comment:
Why? I dont think that makes any more sense because you first have to check that location services are enabled, then you have to check whether location authorization status is one of two values